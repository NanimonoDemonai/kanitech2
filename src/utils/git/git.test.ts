import { simpleGit } from "simple-git";

const git = simpleGit();
describe("git", () => {
  it("should get log", async () => {
    git.init();
    const log = await git.log({
      file: "entries/how_to_mdx_bundling.mdx",
      format: "fuller",
    });
    expect(log).toMatchSnapshot();

    const data = await git.show(
      "b1426f894423fff7b9cf679b566a1553508c15cd:entries/how_to_mdx_bundling.mdx"
    );
    expect(data).toMatchSnapshot();
  });
});
