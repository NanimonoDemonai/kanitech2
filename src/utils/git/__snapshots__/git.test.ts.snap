// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git should get log 1`] = `
{
  "all": [
    {
      "author_email": "masa9nao5.n4.18@gmail.com",
      "author_name": "nanimono_demonai",
      "body": "",
      "date": "2022-11-20T17:50:18+09:00",
      "hash": "b1426f894423fff7b9cf679b566a1553508c15cd",
      "message": "コミットする",
      "refs": "HEAD -> git_test",
    },
    {
      "author_email": "masa9nao5.n4.18@gmail.com",
      "author_name": "nanimono_demonai",
      "body": "",
      "date": "2022-11-19T16:10:19+09:00",
      "hash": "eb50028da73216e8a46927c92f1272fd672aa960",
      "message": "pidでmdxをバンドルする",
      "refs": "",
    },
    {
      "author_email": "masa9nao5.n4.18@gmail.com",
      "author_name": "nanimono_demonai",
      "body": "",
      "date": "2022-11-16T22:03:14+09:00",
      "hash": "2b960b9fb1ae0b86b61d0efed56f8c484a949740",
      "message": "mdxをコンパイルすることができるところまで",
      "refs": "origin/main",
    },
  ],
  "latest": {
    "author_email": "masa9nao5.n4.18@gmail.com",
    "author_name": "nanimono_demonai",
    "body": "",
    "date": "2022-11-20T17:50:18+09:00",
    "hash": "b1426f894423fff7b9cf679b566a1553508c15cd",
    "message": "コミットする",
    "refs": "HEAD -> git_test",
  },
  "total": 3,
}
`;

exports[`git should get log 2`] = `
"---
title: "どのようにMDXをバンドルしているか"
tags: [mdx, nextjs, 解説記事]
---

# テスト1

## Sanguine ambae

Lorem markdownum ipsa ingemit messes te moenia meas neque, se nec. Obvertere
iuvenes faciemque. [Somnus](http://hoc.io/geminos.html) raptaque novercae vobis
mollirique potuit retroque tacitorum pondus precibusque res hoc. Adfixa velat
sumpsit traherent removerat quater ara populos. Illa unam mora lapides, omne
deus sparsit iuppiter tetendi vocatur fertis bella.

Per quae patriorum talibus omnes. Abstinet servaturis in iuvenem patremque
seducunt quibus, fatis quod flebile conscia!

## Fundae cultique conducat propulit

Vidisse in nomenque armis. Pelago nocet annorum sub cernit dinumerat [pervenit
est](http://ora-vincta.net/) antri imi magna, stabula et, semine: in: precor.
Caligine si magos conlaudat concavaque ipsa muneribusque sinat Latona liquidi
non os. Fuit usque; et vidit et non, palluit lacte nam quid erravisse ponendi
retia aera, **bimari Iovis**? Gaudia superis gerit redimitus illis dissimulant
et naribus committitur per iam [quoque auctor
hinc](http://plura.org/aperto.php).

> Ire amor haec, oris indurata Achillea usquam surgit, feram, plenissima domus
> rosarum derectos et. Mixta tum ignaroque secreta difficilis cornua hostem. In
> dis tunc est saecula, pollentibus tura caesariem fontesque.

\\\`\\\`\\\`javascript
alert("hello");
\\\`\\\`\\\`

## Omnia ingeniis utque

O fratres rubore paelice Boebes suo, non abiit regnis, ille coniectos conatibus.
Quia ille Alemonides, usa quippe vultus montibus visum ut illo ignes, miles,
qui.

## Praeferri et oscula medio tempus ferrugine litora

Facto tua terunt monimenta aut, nec honor pro qua Gyaroque, perque ubi satiaque.
Fulserunt sibila sinu concedimus, probavit sitim. Aut aspice paludes, maturus
centum damno. Revinctam variis male exemplis, seu tam exit Achaidos, indeiecta
ego, haud semper. Me raptam fronte cum hactenus aram latentis capiat pendet
sibi.

Ora ingens, dedisse nullaque movi opem furibunda tecum, ingenti? Rutuli Iuno
illa, Glaucus. Sinebat ala atque dis magna, ac in praecordia ignesque signa in
haec. Haut pro, cum ludit corpus; cum attollit
[navigat](http://cancer.org/melumina.aspx) occiduae ferinae certo _superi_, aut
ter!

\`"
`;
